arch/posix/CMakeLists.txt

```diff
@@ -1,5 +1,4 @@
 zephyr_compile_options(
-  -fno-freestanding
```

Comment:
Was this superfluous?

How does this not break anything?

This should be in it's own commit.

Reply:
clang does not has this option and as far as tests go and I know it does not break anything.